libtock/util/streaming_process_slice.c

```diff
@@ -101,43 +94,55 @@ returncode_t streaming_process_slice_get_and_swap(
 
 returncode_t streaming_process_slice_deinit(
   streaming_process_slice_state_t* state,
-  uint8_t**                        buffer,
-  size_t*                          size) {
-  uint8_t* ret_buffer;
-  size_t ret_size;
+  uint8_t**                        buffer_a,
+  size_t*                          size_a,
+  uint8_t**                        buffer_b,
+  size_t*                          size_b) {
+  uint8_t* ret_buffer_a, * ret_buffer_b;
+  size_t ret_size_a, ret_size_b;
```

Comment:
Please don't do multiple variable initializations on a single line. Especially with pointers. It's way too easy to make a mistake.

And is there any reason to not give these default values of NULL and 0 respectively? Then we could again remove the default case in the if statement.

Reply:
Makes sense. Changed to default initializers!